docs(fields): add missing validation methods in jsdoc and improve fields documentation (#958)
Refinitiv/refinitiv-ui: documents/src/pages/elements/search-field.md

`ef-search-field` has validation logic similar to a native input. When a user types the invalid value into the control, error style will be shown to notify the user. However, if the control is being initialised with an invalid value, `reportValidity()` must be called to ensure the error style is applied.

::
```javascript
::search-field::
const searchField = document.querySelector('ef-search-field');
const button = document.querySelector("ef-button");

button.addEventListener('tap', () => {
  searchField.reportValidity();
})
```
```html
<ef-search-field pattern="[a-z]" value="1"></ef-search-field>
<ef-button>Submit</ef-button>
```
::

### Displaying error messages

> Whenever input is invalid, the error attribute will be added to the element. You can use the `error` property to check if input is currently in the error state.

Reviewer: ```suggestion
Whenever input is invalid, the `error` attribute will be added to the element. You can use the `error` property to check if input is currently in the error state. Note that, if input is initialised with invalid value, `reportValidity()` must be called first as described in [Input Validation](/elements/search-field#input-validation).
```

See the [Input Length](/elements/search-field#input-length) example below for more detail.

## Input length

The `maxlength` attribute limits the number of characters that users can enter and the `minlength` attribute sets the minimum number of characters required. `ef-search-field` will show error styles if a condition is not met.

> @> Constraint validation is only applied when the value is changed by the user. [See input search](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search#maxlength).

Reviewer: ```suggestion
@> `maxlength` and `minlength` constraint validations are only applied when the value is changed by the user. [See input search](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search#maxlength).
```

::
```javascript
::search-field::
const searchField = document.querySelector("ef-search-field");
const errorText = document.getElementById("error-text");

searchField.addEventListener("blur", () => {
  if (searchField.error) {
    errorText.innerHTML = "Value length must be between 2 - 4 characters";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = searchField.error ? "Value length must be between 2 - 4 characters." : "";
```

const searchField = document.querySelector("ef-search-field");
const errorText = document.getElementById("error-text");

searchField.addEventListener("blur", () => {
  if (searchField.error) {
    errorText.innerHTML = "Value length must be between 2 - 4 characters";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = searchField.error ? "Value length must be between 2 - 4 characters." : "";
```

});

searchField.addEventListener("input", () => {
  if (!searchField.error) {
    errorText.innerHTML = "";
  }
});
```

```typescript
import { SearchField } from '@refinitiv-ui/elements/search-field';

> const searchField = document.querySelector("ef-search-field") as SearchField;

Reviewer: ```suggestion
const searchField = document.querySelector<SearchField>("ef-search-field");
```

const errorText = document.getElementById("error-text");

searchField?.addEventListener("blur", () => {
  if (!errorText) {
    return;
  }
  if (searchField?.error) {
    errorText.innerHTML = "Value length must be between 2 - 4 characters";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = searchField.error ? "Value length must be between 2 - 4 characters." : "";
```

::search-field::
const searchField = document.querySelector("ef-search-field");
const errorText = document.getElementById("error-text");

searchField.addEventListener("blur", () => {
  if (searchField.error) {
    errorText.innerHTML = "Value must be uppercase letters and has 2 - 5 characters.";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = searchField.error ? "Value must be uppercase letters and has 2 - 5 characters." : "";
```

```javascript
const searchField = document.querySelector("ef-search-field");
const errorText = document.getElementById("error-text");

searchField.addEventListener("blur", () => {
  if (searchField.error) {
    errorText.innerHTML = "Value must be uppercase letters and has 2 - 5 characters.";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = searchField.error ? "Value must be uppercase letters and has 2 - 5 characters." : "";
```

});

searchField.addEventListener("input", () => {
  if (!searchField.error) {
    errorText.innerHTML = "";
  }
});
```

```typescript
import { SearchField } from '@refinitiv-ui/elements/search-field';

> const searchField = document.querySelector("ef-search-field") as SearchField;

Reviewer: ```suggestion
const searchField = document.querySelector<SearchField>("ef-search-field");
```

const errorText = document.getElementById("error-text");

searchField?.addEventListener("blur", () => {
  if (!errorText) {
    return;
  }
  if (searchField?.error) {
    errorText.innerHTML = "Value must be uppercase letters and has 2 - 5 characters.";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: let's simplify with this:
```
errorText.innerHTML = searchField.error ? "Value must be uppercase letters and has 2 - 5 characters." : "";
```
